internal/compression/deflate.go

```diff
@@ -31,33 +33,21 @@ func (c *deflateDecompressor) Read(bytes []byte) (int, error) {
 	return c.reader.Read(bytes)
 }
 func (c *deflateDecompressor) Reset(rdr io.Reader) error {
-	resetter, ok := c.reader.(flate.Resetter)
-	if !ok {
-		// This should never happen as the returned type from flate should always
-		// implement Resetter, but the check is here as a safeguard just in case.
-		// This error would be a very exceptional / unexpected occurrence.
-		return errors.New("deflate reader is not able to be used as a resetter")
-	}
-	// Mimics NewReader internal logic, which initializes the internal dict to nil
-	return resetter.Reset(rdr, nil)
+	var err error
+	c.reader, err = zlib.NewReader(rdr)
+	return err
```

Comment:
This could leave `c.reader` as nil. Maybe it would be safest to do something like so:
```suggestion
	decomp, err := zlib.NewReader(rdr)
	if err != nil {
		c.reader = errDecompressor{err: err}
		return err
	}
	c.reader = decomp
	return nil
```

An alternative would be to put a nil check for `c.reader` in the other methods, to return reasonable errors instead of letting them panic.